contracts/CurrencyNetworkRegistry.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity ^0.5.8;
+
+import "./CurrencyNetworkInterface.sol";
+import "./lib/ERC165Query.sol";
+
+
+contract CurrencyNetworkRegistry is ERC165Query {
+
+    struct CurrencyNetworkMetadata {
+        address author;
+        string name;
+        string symbol;
+        uint8 decimals;
+    }
+
+    mapping (address => CurrencyNetworkMetadata) internal currencyNetworks;
+    address[] internal registeredCurrencyNetworks;
+
+    event CurrencyNetworkAdded(
+        address indexed _address,
+        address _author,
+        string _name,
+        string _symbol,
+        uint8 _decimals
+    );
+
+    function addCurrencyNetwork(address _address) external {
+        CurrencyNetworkInterface network;
+
+        require(
+            currencyNetworks[_address].author == address(0) &&
```

Comment:
I am not sure we want currency networks to be registered only once. I believe one goal would be for a user to ask for `every currency networks registered by the TL Foundation`. If they can be registered only once, then Bernd could register every CurrencyNetworks deployed by the TL Foundation by himself and prevent the TL Foundation from ever registering them.

Reply:
I would add a `mapping (address => address)` that stores who registers what network and update it when someone attempts to register a network that is already registered.

Reply:
Sounds reasonable. Added with https://github.com/trustlines-protocol/contracts/pull/234/commits/d71646551499aaeaa626c9debfa4054bd186f92e